examples/mixture/plot_gmm_init.py

```diff
@@ -0,0 +1,155 @@
+
+See :ref:`gmm` for more information on the estimator.
+
+Here we generate some sample data with four easy to identify clusters.The
+purpose of this example is to show the four different methods for the
+initialization parameter *init_param*.
+
+- *kmeans* - performs kmeans clustering on the dataset to produce the initial
+  centers. This will be the most resource intensive initialization but will
+  often produce intitial means that are very close to the converged gmm model.
+- *k-means++* - uses the initialization of kmeans clustering (called k-means++)
+  to use input data points as the initial centers but to pick them such that
+  they are sufficiently far apart from each other.
+- *rand_data* - uses randomly selected data points as the initial centers. This
+  is a fast method but can potentially lead to unstable results if all of the
+  selected points are close together.
+- *random* - Is the simplest method. It uses centers that are close to the mean
+  of the data. This can be the slowest initialization to converge.
+
+Orange diamonds represent the initialization centers for the gmm generated by
+the *ini_param*. The rest of the data is represented as crosses and the
+colouring represents the eventual associated classification after the gmm has
+finished.
+"""
+
+
+# Author Gordon Walsh (github: g-walsh) gordon.p.walsh@gmail.com
+# data generation code from https://jakevdp.github.io/
+
+import matplotlib.pyplot as plt
+import numpy as np
+from sklearn.cluster import KMeans
+from sklearn.mixture import GaussianMixture
+from sklearn.utils.extmath import row_norms
+from sklearn.cluster.k_means_ import _k_init
+from sklearn.datasets.samples_generator import make_blobs
+
+print(__doc__)
+
+# Generate some data
+
+X, y_true = make_blobs(n_samples=400, centers=4,
+                       cluster_std=0.60, random_state=0)
+X = X[:, ::-1]
+
+n_samples = 400
+n_components = 4
+x_squared_norms = row_norms(X, squared=True)
+
+
+def genGmm(init_params, seed=None):
+    r = np.random.RandomState(seed)
+    if init_params == 'kmeans':
+        init_means = calculateMeans(kmeansMean(r))
+    elif init_params == 'random':
+        init_means = calculateMeans(randMean(r))
+    elif init_params == 'rand_data':
+        init_means = calculateMeans(randPointMean(r))
+    elif init_params == 'k-means++':
+        init_means = calculateMeans(kmeansPlusPlus(r))
+    else:
+        raise ValueError("Unimplemented initialisation method '%s'"
+                         % init_params)
+
+    gmm = GaussianMixture(n_components=4, means_init=init_means, tol=1e-9,
+                          max_iter=2000, random_state=r).fit(X)
+    labels = gmm.predict(X)
+    # plt.scatter(X[:, 0], X[:, 1], c=labels, s=40, cmap='viridis')
+    # plt.scatter(init_means[:, 0], init_means[:, 1], s=100,
+    #             marker='x', c='black')
+    return labels, init_means, seed, init_params
+    # plt.show()
+
+
+def kmeansMean(r):
```

Comment:
please avoid camel case in function names. underscores are your friend in PEP8 Python